File: 04-a-big-barbecue/app/src/app/components/big-barbecue/barbeque-cost-result.ts
import { BarbecueItem } from "./barbecue-item";

export class BarbequeCostResult {
  name: string;
  barbequeItemsTotalCost: number;
  costPerPerson: number;
  remainingBudget: number;
  barbecueItems: BarbecueItem;
}

File: 04-a-big-barbecue/api/Entities/BarbequeCostResult.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace BarbecueCost.Api.Entities
{
    public class BarbequeCostResult
    {
        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("barbequeItemsTotalCost")]
        public double? BarbequeItemsTotalCost { get; set; }

        [JsonProperty("costPerPerson")]
        public double? CostPerPerson { get; set; }

        [JsonProperty("remainingBudget")]
        public double? RemainingBudget { get; set; }

        [JsonProperty("barbecueItems")]
        public List<BarbecueItem> BarbecueItems { get; set; }
    }
}